packages/data/src/components/registry-provider.js

```diff
@@ -0,0 +1,26 @@
+/**
```

Comment:
There's a fair bit of inconsistency in how we're creating components here and with relation to conventions of other modules.

1. `withDispatch` and `withSelect` are in their own files, but `withRegistry` is combined into the provider file
   - If we want a pattern of component-per-file, we should be strictly consistent. Case in point: I searched `with-registry` on this page expecting to find the file where this component was defined, but it did not exist.
2. Other modules we have folder pattern `components/registry-provider/index.js` .
   - We may argue this is the "next step of simplest", but if we're already to the point of splitting by component, we may as well strive for consistency. It also provides an easier pattern for ensuring we're responsible for including README per component.
3. Other modules we place higher-order components within a nested `higher-order` folder.
   - While the naming/placement could be subject to debate, the point is: (a) consistency, (b) accurately reflecting that these aren't actually components in their own right, and can't be used directly, only as an enhancer to another component.

Reply:
Fair points:

I'm leaning towards:

 `components/registry-provider/index.js` / `components/with-select/index.js`. I personally don't think the `higher-order` folder is necessary because sometimes we need both from the same file `createContext` calls.

I can extract `with-registry` but my idea was to avoid exporting the `Consumer` entirely. It's minor though. but even if we separate withRegistry we still need to export two components because `createContext` produces two components. So not certain how to go with this.

Reply:
>even if we separate withRegistry we still need to export two components because `createContext` produces two components. So not certain how to go with this.

I think it's fine to have an exception for `createContext`, mostly out of necessity.